cpp/include/partition_manager.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/comms/comms.hpp>
+#include <raft/handle.hpp>
+
+#include <cassert>
+#include <cmath>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace cugraph {
+namespace partition_2d {
+template <typename from_t>
+std::string to_string(from_t const& value)
+{
+  std::stringstream ss;
+  ss << value;
+  return ss.str();
+}
+
+// default key-naming mechanism:
+//
+struct key_naming_t {
+  // simplified key (one per all row subcomms / one per all column sub-comms):
+  //
+  key_naming_t(void)
+    : row_suffix_(std::string("_p_row")),
+      col_suffix_(std::string("_p_col")),
+      prefix_(std::string("comm")),
+      name_(prefix_)
+  {
+  }
+
+  std::string col_name(void) const { return name_ + col_suffix_; }
+
+  std::string row_name(void) const { return name_ + row_suffix_; }
+
+ private:
+  std::string const row_suffix_;
+  std::string const col_suffix_;
+  std::string const prefix_;
+  std::string name_;
```

Comment:
It seems like now name_ & prefix_ are redundant. Besides this, this looks good to me.

Reply:
Fixed.